roles/translator/src/proxy/bridge.rs

```diff
@@ -104,7 +142,183 @@ impl Bridge {
             future_jobs: vec![],
             last_p_hash: None,
             target,
-        }
+            first_ph_received: false,
+            request_ids: Id::new(),
+            pool_output_is_set: false,
+            solution_sender,
+        }));
+        match upstream_kind {
+            UpstreamKind::Standard => (),
+            UpstreamKind::WithNegotiator {
+                recv_tp,
+                recv_ph,
+                recv_coinbase_out,
+                send_mining_job,
+                ..
+            } => {
+                // open a channel so that jobs are created and last notify is updated also if no
+                // dowsntream connected
+                self_
+                    .safe_lock(|s| {
+                        s.channel_factory.new_extended_channel(0, 1.0, 0);
+                    })
+                    .unwrap();
+                Self::start_receiving_pool_coinbase_outs(self_.clone(), recv_coinbase_out);
+                Self::start_receiving_new_template(self_.clone(), recv_tp, send_mining_job.clone());
+                Self::start_receiving_new_prev_hash(self_.clone(), recv_ph, send_mining_job);
+            }
+        };
+        self_
+    }
+
+    pub fn start_receiving_pool_coinbase_outs(
+        self_mutex: Arc<Mutex<Self>>,
+        recv: Receiver<(Vec<TxOut>, u64)>,
+    ) {
+        task::spawn(async move {
+            while let Ok((outs, _id)) = recv.recv().await {
+                // TODO MVP3 assuming that only one coinbase is negotiated with the pool not handling
+                // different tokens, in order to do that we can use the out hashmap:
+                // self.tx_outs.insert(id, outs)
+                self_mutex
+                    .safe_lock(|s| {
+                        s.channel_factory.update_pool_outputs(outs);
+                        s.pool_output_is_set = true;
+                    })
+                    .unwrap();
```

Comment:
use handle_result here so the main can die if something bad happens within this task.

Reply:
Maybe I'm missing something but I'm not sure why should I use `handle_result` here? The above can only fail if the `Mutex` is poisoned, that means that this issue is not recoverable and we should just panic and terminate the task. We can poison the Mutex if we panic inside a `safe_lock` and that should be always avoided.

Reply:
For now let's use the PoisonLock error but what I would really love to see is to have closure inside `safe_lock` that can not panic (I saw that there is some lib that that can check if a function can panic or it will never panic).

What I think we should try to achieve is remove every possibility to panic from the function inside `safe_lock`. having something in the signature of `safe_lock` that make the closure arguments not panicable.

Reply:
opened an issue about that https://github.com/stratum-mining/stratum/issues/476